images/virtualization-artifact/pkg/controller/vm/internal/size_policy.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/deckhouse/virtualization-controller/pkg/controller/conditions"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/service"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/vm/internal/state"
+	"github.com/deckhouse/virtualization-controller/pkg/logger"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2/vmcondition"
+)
+
+const nameSizePolicyHandler = "SizePolicyHandler"
+
+func NewSizePolicyHandler() *SizePolicyHandler {
+	return &SizePolicyHandler{
+		service: service.NewSizePolicyService(),
+	}
+}
+
+type SizePolicyHandler struct {
+	service *service.SizePolicyService
+}
+
+func (h *SizePolicyHandler) Handle(ctx context.Context, s state.VirtualMachineState) (reconcile.Result, error) {
+	if s.VirtualMachine().IsEmpty() {
+		return reconcile.Result{}, nil
+	}
+	current := s.VirtualMachine().Current()
+	changed := s.VirtualMachine().Changed()
+
+	addAllUnknown(changed, vmcondition.TypeSizingPolicyMatched)
+
+	if isDeletion(current) {
+		return reconcile.Result{}, nil
+	}
+
+	vmClass, err := s.Class(ctx)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
helper.FetchObject is used by the s.Class method. No need to handle "not found" error